.storybook/main.js

```diff
@@ -56,17 +57,28 @@ module.exports = /** @type {Omit<StorybookConfig,'typescript'|'babel'>} */ ({
     }
 
     if (config.module && config.module.rules) {
-      overrideDefaultBabelLoader(/** @type {import("webpack").RuleSetRule[]} */ (config.module.rules));
-
-      config.module.rules.unshift({
+      const customLoaderPath = path.resolve(__dirname, './custom-loader.js');
+      /**
+       * @type {import("webpack").RuleSetRule}
+       */
+      const codesandboxRule = {
         test: /\.stories\.tsx$/,
+        include: /src/,
         exclude: /node_modules/,
         use: {
           loader: 'babel-loader',
           options: {
-            plugins: [[require('storybook-addon-export-to-codesandbox').babelPlugin, getCodesandboxBabelOptions()]],
+            plugins: [[exportToCodesandboxAddon.babelPlugin, getCodesandboxBabelOptions()]],
+            // NOTE: this is important to add, every-time you'll be adding some new rule that includes `babel-loader` this needs to be added !
+            customize: customLoaderPath,
           },
         },
+      };
+
+      config.module.rules.unshift(codesandboxRule);
```

Comment:
why do we need to have it as first rule ? cc @ling1726

Reply:
can't remember, I don't think it needs to be tho

Reply:
I revamp the implementation a bit and more "know how" within comments